app/src/main/java/org/schabi/newpipe/local/dialog/PlaylistAppendDialog.java

```diff
@@ -154,17 +155,19 @@ private void onPlaylistSelected(@NonNull final LocalPlaylistManager manager,
 
         final Toast successToast = Toast.makeText(getContext(), toastText, Toast.LENGTH_SHORT);
 
-        if (playlist.thumbnailUrl
-                .equals("drawable://" + R.drawable.placeholder_thumbnail_playlist)) {
-            playlistDisposables.add(manager
-                    .changePlaylistThumbnail(playlist.uid, streams.get(0).getThumbnailUrl(), false)
-                    .observeOn(AndroidSchedulers.mainThread())
-                    .subscribe(ignored -> successToast.show()));
-        }
-
         playlistDisposables.add(manager.appendToPlaylist(playlist.uid, streams)
                 .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(ignored -> successToast.show()));
+                .subscribe(ignored -> {
+                    successToast.show();
+
+                    if (playlist.thumbnailUrl.equals(PlaylistEntity.DEFAULT_THUMBNAIL)) {
+                        playlistDisposables.add(manager
+                                .changePlaylistThumbnail(playlist.uid, streams.get(0).getUid(),
+                                        false)
+                                .observeOn(AndroidSchedulers.mainThread())
+                                .subscribe(ignore -> successToast.show()));
+                    }
```

Comment:
Why was this moved here?

Reply:
The reason for this change is addressed in [this comment](https://github.com/TeamNewPipe/NewPipe/pull/9755#discussion_r1102103768).